src/Main.java

```diff
@@ -0,0 +1,29 @@
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+public class Main {
+    public static void main(String[] args) {
+        if(args.length != 2) {
+            System.err.println("Missing required parameters: <pattern> <file>");
+            return;
+        }
+
+        Pattern pattern = Pattern.compile(args[0]);
+
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(args[1]));
```

Comment:
Hard to read, break into descriptive steps

Reply:
Likely better to not have an extra reference. All reads should go through BufferedReader.